Lib/test/test_threading.py

```diff
@@ -2130,6 +2130,11 @@ def test_set_name(self):
 
             # Test long non-ASCII name (truncated)
             "x" * (limit - 1) + "é€",
+
+            # Test long non-BMP names (truncated) creating surrogate pairs
+            # on Windows
+            "x" * (limit - 1) + "\U0010FFFF",
+            "x" * (limit - 2) + "\U0010FFFF" * 2,
```

Comment:
Add also tests for (for both Windows and non-Windows):
"x"  + "\U0001f40d" * limit
"xx"  + "\U0001f40d" * limit
"xxx"  + "\U0001f40d" * limit
"xxxx"  + "\U0001f40d" * limit

It tests that the limit is for the encoded name.

Reply:
Ok, I added these tests.